### ISSUE TITLE
Cannot read property '$reql_type$' of undefined

### ISSUE THREAD

**Issue by jonathanepollack** [neumino/thinky#164]
Every time I try to save a document that has a value with type `Point`, or try to delete a document that is linked to another document with a 'Point' value, I get the error:

```
Possibly unhandled TypeError: Cannot read property '$reql_type$' of undefined
    at /vagrant/node_modules/thinky/lib/document.js:369:77
    at Object.loopKeys (/vagrant/node_modules/thinky/lib/util.js:177:22)
    at Document.__makeSavableCopy (/vagrant/node_modules/thinky/lib/document.js:366:14)
    at /vagrant/node_modules/thinky/lib/document.js:382:54
    at Object.loopKeys (/vagrant/node_modules/thinky/lib/util.js:177:22)
    at model.Document.__makeSavableCopy (/vagrant/node_modules/thinky/lib/document.js:366:14)
    at model.Document._makeSavableCopy (/vagrant/node_modules/thinky/lib/document.js:333:17)
    at /vagrant/node_modules/thinky/lib/document.js:471:25
    at tryCatch2 (/vagrant/node_modules/thinky/node_modules/bluebird/js/main/util.js:51:21)
    at Promise$_resolveFromResolver [as _resolveFromResolver] (/vagrant/node_modules/thinky/node_modules/bluebird/js/main/promise.js:585:13)
```

However, if I change the value to something like `document.location = [lat, lang]` before I save, thinky doesn't seem to care (and will then internally cast my value into a reql Point)...

There doesn't seem to be a hack for dealing with deleting though.

**Comment by neumino** [comment]
@jonathanepollack -- What version of thinky are you running? 
I tried to declare a model with a type `type.point()` and insert a ReQL point (`r.point(1, 1)`) and it works fine for me.

Can you provide a snippet that triggers this error?

**Comment by jonathanepollack** [comment]
Weirdly, I can't reproduce this on delete, so I'm going to close this ticket for now.

**Comment by assafweinberg** [comment]
I'm seeing similar behavior.  I can successfully update a document that has a point in it by calling .save(), but if I try to do a .saveAll() on a document that has a many to one relationship with a document that has a point, I get the error above.  For example:

```
var Account = thinky.createModel("Account", {
  id: type.string(),
  contractor_id: type.string(),
})

var Contractor = thinky.createModel("Contractor", {
  id: type:string(),
  location: type.point()
})

Contractor.hasMany(Account, 'accounts', 'id', 'contractor_id');
Account.belongsTo(Contractor, 'contractor', 'contractor_id', 'id');

//Trying to arbitrarily add a contractor to an account
 var acct;
 var contr;
  Models.Account
    .getJoin()
    .run()
    .then(function(accounts){
      acct = accounts[0];
      return Models.Contractor.run()
    })
    .then(function(contractors){
      contr = contractors[0];
      console.log(acct, contr);
      acct.contractor = contr;
      
      //Uncommenting the following line or removing the geoLocation attribute will cause the statement  to succeed without error
      //contr.geoLocation = [1,2];

      return acct.saveAll();
    })
```

This is my first attempt at using Thinky.  Let me know if I'm doing something obviously wrong here.

**Comment by neumino** [comment]
@assafweinberg -- Sorry I did a bit of refactoring and introduced a bug on saving raw ReQL points.
I just released a fix in 1.16.12. Let me know if it works for you
